### PR TITLE
[ZHF] perlPackages.TextBibTeX: use lib instead of lib64 on aarch64

### LINKED ISSUE #122714: biber fails to build on aarch64
**Describe the bug**
https://hydra.nixos.org/build/142042799/nixlog/2

**To Reproduce**
```
nix-build -A biber --argstr system aarch64-linux
```
(works if you have `boot.binfmt.emulatedSystems = [ "aarch64-linux" ]` in your `configuration.nix`)

**Notify maintainers**

@ttuegel

### PR DESCRIPTION
**Pull request by xworld21**
###### Motivation for this change
Build failure https://hydra.nixos.org/build/142787205

This is absolutely a shot in the dark, as I don't have anywhere to test this.

Is it ok to let OfBorg build it?

PS:  I suppose that `Text::BibTeX` should actually ask `DynaLoader` what is the standard library folder, or something like that, instead of guessing `lib` vs `lib64`. I will report this upstream if this is indeed the source of the problem.

Cc @veprbl

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/perl-packages.nix [hunk 1]
```diff
@@ -21802,7 +21802,10 @@ let
       description = "Interface to read and parse BibTeX files";
       license = with lib.licenses; [ artistic1 gpl1Plus ];
     };
-  };
+  } // lib.optionalAttrs stdenv.isAarch64 {
+    # libbtparse.so: cannot open shared object file (aarch64 only)
+    patches = [ ../development/perl-modules/TextBibTeX-use-lib-on-aarch64.patch ];
+  });
```
Comment (SuperSandro2000) [inline pkgs/top-level/perl-packages.nix:21808]: Please do not apply this patch conditionally. It does not break amd64 and would only bit rot over time.

Comment (xworld21) [reply pkgs/top-level/perl-packages.nix]: very good point

**Comment by dotlambda** [comment]
@ofborg build biber

**Comment by dotlambda** [comment]
Result of `nixpkgs-review pr 122567` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>manim</li>
    <li>texlive.combined.scheme-full</li>
  </ul>
</details>
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>biber</li>
    <li>perl530Packages.TextBibTeX</li>
    <li>perl532Packages.TextBibTeX</li>
  </ul>
</details>

**Comment by dotlambda** [comment]
I guess texlive needs some more fixing but we can do that in a separate PR:
```
Testing restricted mode for {,r}epstopdf
!!! Error: Option forbidden in restricted mode: --gscmd
This is LuaTeX, Version 1.13.0 (Web2C 2021/NixOS.org)
 restricted system commands enabled.

(/nix/store/vyx903ybpi1awc6282064fbqff208wsj-texlive-context-58167/scripts/cont
ext/stubs/unix/mtxrun
! You can't use `macro parameter character #' in vertical mode.
l.1 #
   !/usr/bin/env texlua
?
! Emergency stop.
l.1 #
   !/usr/bin/env texlua
!  ==> Fatal error occurred, no output PDF file produced!
Transcript written on mtxrun.log.
```

> I suppose that `Text::BibTeX` should actually ask `DynaLoader` what is the standard library folder, or something like that, instead of guessing `lib` vs `lib64`. I will report this upstream if this is indeed the source of the problem.

That would be great!

**Comment by xworld21** [comment]
So this actually works!

Although, I have just read `move-lib64.sh`, and it looks like this bug just shouldn't happen. I am quite puzzled now. Oh well, if it works...

> > I will report this upstream if this is indeed the source of the problem.
> 
> That would be great!

Will do!

**Pull request merged**